pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/MhloLegalizeUtils.cpp

```diff
@@ -130,44 +150,142 @@ template llvm::Optional<Value> getConstTensor<int64_t>(
     ArrayRef<int64_t> vec,
     ArrayRef<int64_t> shape);
 
+std::vector<Value> getDimSizesOfTensor(
+    PatternRewriter& rewriter,
+    Operation* op,
+    Value value,
+    ArrayRef<int64_t> inpDims) {
+  auto currentKnowledge = ValueKnowledge::getKnowledgeFromType(value.getType());
+  std::vector<Value> dimSizes;
+  if (!currentKnowledge.hasRank) {
```

Comment:
nit: maybe return a `llvm::Optional<std::vector<Value>>` is better? I see here return directly while the callee can not know it's a failure case actually.

Reply:
would empty `dimSizes` get mixed up with scalar type?

Reply:
I added codes to emit an error message if they meet illegal usages.